test_heliographic_grid_annotations warning about memory access

One of the warnings that https://github.com/sunpy/sunpy/pull/3648 has revealed is below. I spent about half a day trying to work out what was going on, but to no avail. Full error message below:

```python
sts/test_plotting.py F                                                                                                                                                                       [100%]

=================================================================================================== FAILURES ===================================================================================================
______________________________________________________________________________________ test_heliographic_grid_annotations ______________________________________________________________________________________

coords = <UnitSphericalRepresentation (lon, lat) in deg
    [(nan, nan), (nan, nan), (nan, nan), ..., (nan, nan), (nan, nan),
     (nan, nan)]>
frame = <HeliographicCarrington Frame (obstime=2017-01-27T16:54:20.000, observer=<HeliographicStonyhurst Coordinate (obstime=2017-01-27T16:54:20.000): (lon, lat, radius) in (deg, deg, AU)
    (0., -5.70644239, 0.98482348)>)>
units = [None, None, None], init_kwargs = {}

    def _parse_coordinate_arg(coords, frame, units, init_kwargs):
        """
        Single unnamed arg supplied.  This must be:
        - Coordinate frame with data
        - Representation
        - SkyCoord
        - List or tuple of:
          - String which splits into two values
          - Iterable with two values
          - SkyCoord, frame, or representation objects.
    
        Returns a dict mapping coordinate attribute names to values (or lists of
        values)
        """
        from .sky_coordinate import SkyCoord
    
        is_scalar = False  # Differentiate between scalar and list input
        # valid_kwargs = {}  # Returned dict of lon, lat, and distance (optional)
        components = {}
        skycoord_kwargs = {}
    
        frame_attr_names = list(frame.representation_component_names.keys())
        repr_attr_names = list(frame.representation_component_names.values())
        repr_attr_classes = list(frame.representation_type.attr_classes.values())
        n_attr_names = len(repr_attr_names)
    
        # Turn a single string into a list of strings for convenience
        if isinstance(coords, str):
            is_scalar = True
            coords = [coords]
    
        if isinstance(coords, (SkyCoord, BaseCoordinateFrame)):
            # Note that during parsing of `frame` it is checked that any coordinate
            # args have the same frame as explicitly supplied, so don't worry here.
    
            if not coords.has_data:
                raise ValueError('Cannot initialize from a frame without coordinate data')
    
            data = coords.data.represent_as(frame.representation_type)
    
            values = []  # List of values corresponding to representation attrs
            repr_attr_name_to_drop = []
            for repr_attr_name in repr_attr_names:
                # If coords did not have an explicit distance then don't include in initializers.
                if (isinstance(coords.data, UnitSphericalRepresentation) and
                        repr_attr_name == 'distance'):
                    repr_attr_name_to_drop.append(repr_attr_name)
                    continue
    
                # Get the value from `data` in the eventual representation
                values.append(getattr(data, repr_attr_name))
    
            # drop the ones that were skipped because they were distances
            for nametodrop in repr_attr_name_to_drop:
                nameidx = repr_attr_names.index(nametodrop)
                del repr_attr_names[nameidx]
                del units[nameidx]
                del frame_attr_names[nameidx]
                del repr_attr_classes[nameidx]
    
            if coords.data.differentials and 's' in coords.data.differentials:
                orig_vel = coords.data.differentials['s']
                vel = coords.data.represent_as(frame.representation_type, frame.get_representation_cls('s')).differentials['s']
                for frname, reprname in frame.get_representation_component_names('s').items():
                    if (reprname == 'd_distance' and
                            not hasattr(orig_vel, reprname) and
                            'unit' in orig_vel.get_name()):
                        continue
                    values.append(getattr(vel, reprname))
                    units.append(None)
                    frame_attr_names.append(frname)
                    repr_attr_names.append(reprname)
                    repr_attr_classes.append(vel.attr_classes[reprname])
    
            for attr in frame_transform_graph.frame_attributes:
                value = getattr(coords, attr, None)
                use_value = (isinstance(coords, SkyCoord) or
                             attr not in coords._attr_names_with_defaults)
                if use_value and value is not None:
                    skycoord_kwargs[attr] = value
    
        elif isinstance(coords, BaseRepresentation):
            if coords.differentials and 's' in coords.differentials:
                diffs = frame.get_representation_cls('s')
                data = coords.represent_as(frame.representation_type, diffs)
                values = [getattr(data, repr_attr_name) for repr_attr_name in repr_attr_names]
                for frname, reprname in frame.get_representation_component_names('s').items():
                    values.append(getattr(data.differentials['s'], reprname))
                    units.append(None)
                    frame_attr_names.append(frname)
                    repr_attr_names.append(reprname)
                    repr_attr_classes.append(data.differentials['s'].attr_classes[reprname])
    
            else:
                data = coords.represent_as(frame.representation_type)
                values = [getattr(data, repr_attr_name) for repr_attr_name in repr_attr_names]
    
        elif (isinstance(coords, np.ndarray) and coords.dtype.kind in 'if' and
              coords.ndim == 2 and coords.shape[1] <= 3):
            # 2-d array of coordinate values.  Handle specially for efficiency.
            values = coords.transpose()  # Iterates over repr attrs
    
        elif isinstance(coords, (Sequence, np.ndarray)):
            # Handles list-like input.
    
            vals = []
            is_ra_dec_representation = ('ra' in frame.representation_component_names and
                                        'dec' in frame.representation_component_names)
            coord_types = (SkyCoord, BaseCoordinateFrame, BaseRepresentation)
            if any(isinstance(coord, coord_types) for coord in coords):
                # this parsing path is used when there are coordinate-like objects
                # in the list - instead of creating lists of values, we create
                # SkyCoords from the list elements and then combine them.
                scs = [SkyCoord(coord, **init_kwargs) for coord in coords]
    
                # Check that all frames are equivalent
                for sc in scs[1:]:
                    if not sc.is_equivalent_frame(scs[0]):
                        raise ValueError("List of inputs don't have equivalent "
                                         "frames: {} != {}".format(sc, scs[0]))
    
                # Now use the first to determine if they are all UnitSpherical
                allunitsphrepr = isinstance(scs[0].data, UnitSphericalRepresentation)
    
                # get the frame attributes from the first coord in the list, because
                # from the above we know it matches all the others.  First copy over
                # the attributes that are in the frame itself, then copy over any
                # extras in the SkyCoord
                for fattrnm in scs[0].frame.frame_attributes:
                    skycoord_kwargs[fattrnm] = getattr(scs[0].frame, fattrnm)
                for fattrnm in scs[0]._extra_frameattr_names:
                    skycoord_kwargs[fattrnm] = getattr(scs[0], fattrnm)
    
                # Now combine the values, to be used below
                values = []
                for data_attr_name, repr_attr_name in zip(frame_attr_names, repr_attr_names):
                    if allunitsphrepr and repr_attr_name == 'distance':
                        # if they are *all* UnitSpherical, don't give a distance
                        continue
                    data_vals = []
                    for sc in scs:
                        data_val = getattr(sc, data_attr_name)
                        data_vals.append(data_val.reshape(1,) if sc.isscalar else data_val)
                    concat_vals = np.concatenate(data_vals)
                    # Hack because np.concatenate doesn't fully work with Quantity
                    if isinstance(concat_vals, u.Quantity):
                        concat_vals._unit = data_val.unit
                    values.append(concat_vals)
            else:
                # none of the elements are "frame-like"
                # turn into a list of lists like [[v1_0, v2_0, v3_0], ... [v1_N, v2_N, v3_N]]
                for coord in coords:
                    if isinstance(coord, str):
                        coord1 = coord.split()
                        if len(coord1) == 6:
                            coord = (' '.join(coord1[:3]), ' '.join(coord1[3:]))
                        elif is_ra_dec_representation:
                            coord = _parse_ra_dec(coord)
                        else:
                            coord = coord1
                    vals.append(coord)  # Assumes coord is a sequence at this point
    
                # Do some basic validation of the list elements: all have a length and all
                # lengths the same
                try:
                    n_coords = sorted(set(len(x) for x in vals))
                except Exception:
                    raise ValueError('One or more elements of input sequence does not have a length')
    
                if len(n_coords) > 1:
                    raise ValueError('Input coordinate values must have same number of elements, found {}'
                                     .format(n_coords))
                n_coords = n_coords[0]
    
                # Must have no more coord inputs than representation attributes
                if n_coords > n_attr_names:
                    raise ValueError('Input coordinates have {} values but '
                                     'representation {} only accepts {}'
                                     .format(n_coords,
                                             frame.representation_type.get_name(),
                                             n_attr_names))
    
                # Now transpose vals to get [(v1_0 .. v1_N), (v2_0 .. v2_N), (v3_0 .. v3_N)]
                # (ok since we know it is exactly rectangular).  (Note: can't just use zip(*values)
                # because Longitude et al distinguishes list from tuple so [a1, a2, ..] is needed
                # while (a1, a2, ..) doesn't work.
                values = [list(x) for x in zip(*vals)]
    
                if is_scalar:
                    values = [x[0] for x in values]
        else:
            raise ValueError('Cannot parse coordinates from first argument')
    
        # Finally we have a list of values from which to create the keyword args
        # for the frame initialization.  Validate by running through the appropriate
        # class initializer and supply units (which might be None).
        try:
            for frame_attr_name, repr_attr_class, value, unit in zip(
                    frame_attr_names, repr_attr_classes, values, units):
>               components[frame_attr_name] = repr_attr_class(value, unit=unit,
                                                              copy=False)

../astropy/astropy/coordinates/sky_coordinate_parsers.py:563: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'astropy.coordinates.angles.Longitude'>, angle = <Longitude [nan, nan, nan, ..., nan, nan, nan] deg>, unit = None, wrap_angle = <Angle 360. deg>, kwargs = {'copy': False}
self = <Longitude [nan, nan, nan, ..., nan, nan, nan] deg>

    def __new__(cls, angle, unit=None, wrap_angle=None, **kwargs):
        # Forbid creating a Long from a Lat.
        if isinstance(angle, Latitude):
            raise TypeError("A Longitude angle cannot be created from "
                            "a Latitude angle.")
        self = super().__new__(cls, angle, unit=unit, **kwargs)
        if wrap_angle is None:
            wrap_angle = getattr(angle, 'wrap_angle', self._default_wrap_angle)
>       self.wrap_angle = wrap_angle

../astropy/astropy/coordinates/angles.py:620: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Longitude [nan, nan, nan, ..., nan, nan, nan] deg>, value = <Angle 360. deg>

    @wrap_angle.setter
    def wrap_angle(self, value):
        self._wrap_angle = Angle(value, copy=False)
>       self._wrap_internal()

../astropy/astropy/coordinates/angles.py:659: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Longitude [nan, nan, nan, ..., nan, nan, nan] deg>

    def _wrap_internal(self):
        """
        Wrap the internal values in the Longitude object. Using the
        :meth:`~astropy.coordinates.Angle.wrap_at` method causes
        recursion.
        """
        # Convert the wrap angle and 360 degrees to the native unit of
        # this Angle, then do all the math on raw Numpy arrays rather
        # than Quantity objects for speed.
        a360 = u.degree.to(self.unit, 360.0)
        wrap_angle = self.wrap_angle.to_value(self.unit)
        wrap_angle_floor = wrap_angle - a360
        self_angle = self.value
        # Do the wrapping, but only if any angles need to be wrapped
        #
        # This invalid catch block can be removed when the minimum numpy
        # version is >= 1.19 (NUMPY_LT_1_19)
        with np.errstate(invalid='ignore'):
            wraps = (self_angle - wrap_angle_floor) // a360
        if np.any(wraps != 0):
>           self -= (wraps * a360) << self.unit

../astropy/astropy/coordinates/angles.py:650: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Longitude [nan, nan, nan, ..., nan, nan, nan] deg>, args = (<ufunc 'subtract'>, '__call__', <Longitude [nan, nan, nan, ..., nan, nan, nan] deg>, <Quantity [nan, nan, nan, ..., nan, nan, nan] deg>)
kwargs = {'out': (<Longitude [nan, nan, nan, ..., nan, nan, nan] deg>,)}

    def __array_ufunc__(self, *args, **kwargs):
>       results = super().__array_ufunc__(*args, **kwargs)

../astropy/astropy/coordinates/angles.py:668: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Longitude [nan, nan, nan, ..., nan, nan, nan] deg>, function = <ufunc 'subtract'>, method = '__call__'
inputs = (<Longitude [nan, nan, nan, ..., nan, nan, nan] deg>, <Quantity [nan, nan, nan, ..., nan, nan, nan] deg>), kwargs = {'out': (array([nan, nan, nan, ..., nan, nan, nan]),)}, converters = [None, None]
unit = Unit("deg"), out = <Longitude [nan, nan, nan, ..., nan, nan, nan] deg>, out_array = array([nan, nan, nan, ..., nan, nan, nan])
arrays = [array([nan, nan, nan, ..., nan, nan, nan]), array([nan, nan, nan, ..., nan, nan, nan])], input_ = array([nan, nan, nan, ..., nan, nan, nan]), converter = None

    def __array_ufunc__(self, function, method, *inputs, **kwargs):
        """Wrap numpy ufuncs, taking care of units.
    
        Parameters
        ----------
        function : callable
            ufunc to wrap.
        method : str
            Ufunc method: ``__call__``, ``at``, ``reduce``, etc.
        inputs : tuple
            Input arrays.
        kwargs : keyword arguments
            As passed on, with ``out`` containing possible quantity output.
    
        Returns
        -------
        result : `~astropy.units.Quantity`
            Results of the ufunc, with the unit set properly.
        """
        # Determine required conversion functions -- to bring the unit of the
        # input to that expected (e.g., radian for np.sin), or to get
        # consistent units between two inputs (e.g., in np.add) --
        # and the unit of the result (or tuple of units for nout > 1).
        converters, unit = converters_and_unit(function, method, *inputs)
    
        out = kwargs.get('out', None)
        # Avoid loop back by turning any Quantity output into array views.
        if out is not None:
            # If pre-allocated output is used, check it is suitable.
            # This also returns array view, to ensure we don't loop back.
            if function.nout == 1:
                out = out[0]
            out_array = check_output(out, unit, inputs, function=function)
            # Ensure output argument remains a tuple.
            kwargs['out'] = (out_array,) if function.nout == 1 else out_array
    
        # Same for inputs, but here also convert if necessary.
        arrays = []
        for input_, converter in zip(inputs, converters):
            input_ = getattr(input_, 'value', input_)
            arrays.append(converter(input_) if converter else input_)
    
        # Call our superclass's __array_ufunc__
>       result = super().__array_ufunc__(function, method, *arrays, **kwargs)
E       DeprecationWarning: Numpy has detected that you (may be) writing to an array with
E       overlapping memory from np.broadcast_arrays. If this is intentional
E       set the WRITEABLE flag True or make a copy immediately before writing.

../astropy/astropy/units/quantity.py:477: DeprecationWarning

During handling of the above exception, another exception occurred:

args = ()
kwargs = {'heliographic_test_map': <sunpy.map.mapbase.GenericMap object at 0x1222b3be0>
SunPy Map
---------
Observatory:		 
Ins...   -0.02115 ],
       [-0.030485, -0.030435, -0.030385, ..., -0.03059 , -0.030535,
        -0.030485]], dtype=float32)}
name = 'sunpy.map.tests.test_plotting.test_heliographic_grid_annotations', fig = <Figure size 1000x800 with 1 Axes>
result_image_loc = PosixPath('figure_test_images/sunpy.map.tests.test_plotting.test_heliographic_grid_annotations.png')

    @pytest.mark.figure
    @wraps(test_function)
    def wrapper(*args, **kwargs):
        if not os.path.exists(hash.HASH_LIBRARY_FILE):
            pytest.xfail(f'Could not find a figure hash library at {hash.HASH_LIBRARY_FILE}')
        # figure_base_dir is a pytest fixture defined on use.
        if figure_base_dir is None:
            pytest.xfail("No directory to save figures to found")
    
        name = "{}.{}".format(test_function.__module__,
                              test_function.__name__)
        # Run the test function and get the figure
        plt.figure()
        fig = test_function(*args, **kwargs)
        if fig is None:
            fig = plt.gcf()
    
        # Save the image that was generated
        figure_base_dir.mkdir(exist_ok=True)
        result_image_loc = figure_base_dir / f'{name}.png'
>       plt.savefig(str(result_image_loc))

sunpy/tests/helpers.py:101: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../matplotlib/lib/matplotlib/pyplot.py:842: in savefig
    res = fig.savefig(*args, **kwargs)
../matplotlib/lib/matplotlib/figure.py:2307: in savefig
    self.canvas.print_figure(fname, **kwargs)
../matplotlib/lib/matplotlib/backend_bases.py:2135: in print_figure
    result = print_method(
../matplotlib/lib/matplotlib/backends/backend_agg.py:507: in print_png
    FigureCanvasAgg.draw(self)
../matplotlib/lib/matplotlib/backends/backend_agg.py:406: in draw
    self.figure.draw(self.renderer)
../matplotlib/lib/matplotlib/artist.py:41: in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
../matplotlib/lib/matplotlib/figure.py:1861: in draw
    mimage._draw_list_compositing_images(
../matplotlib/lib/matplotlib/image.py:132: in _draw_list_compositing_images
    a.draw(renderer)
../astropy/astropy/visualization/wcsaxes/core.py:458: in draw
    super().draw(renderer, **kwargs)
../matplotlib/lib/matplotlib/artist.py:41: in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
../matplotlib/lib/matplotlib/cbook/deprecation.py:410: in wrapper
    return func(*inner_args, **inner_kwargs)
../matplotlib/lib/matplotlib/axes/_base.py:2746: in draw
    mimage._draw_list_compositing_images(renderer, self, artists)
../matplotlib/lib/matplotlib/image.py:132: in _draw_list_compositing_images
    a.draw(renderer)
../astropy/astropy/visualization/wcsaxes/core.py:45: in draw
    self.axes.draw_wcsaxes(renderer)
../astropy/astropy/visualization/wcsaxes/core.py:415: in draw_wcsaxes
    coord._draw_grid(renderer)
../astropy/astropy/visualization/wcsaxes/coordinate_helpers.py:567: in _draw_grid
    self._update_ticks()
../astropy/astropy/visualization/wcsaxes/coordinate_helpers.py:633: in _update_ticks
    coord_range = self.parent_map.get_coord_range()
../astropy/astropy/visualization/wcsaxes/coordinates_map.py:155: in get_coord_range
    return find_coordinate_range(self._transform,
../astropy/astropy/visualization/wcsaxes/coordinate_range.py:51: in find_coordinate_range
    world = transform.transform(np.vstack([xp.ravel(), yp.ravel()]).transpose())
../matplotlib/lib/matplotlib/transforms.py:1422: in transform
    res = self.transform_affine(self.transform_non_affine(values))
../matplotlib/lib/matplotlib/transforms.py:2353: in transform_non_affine
    return self._b.transform_non_affine(
../astropy/astropy/visualization/wcsaxes/transforms.py:114: in transform
    c_in = SkyCoord(UnitSphericalRepresentation(x_in, y_in),
../astropy/astropy/coordinates/sky_coordinate.py:312: in __init__
    skycoord_kwargs, components, info = _parse_coordinate_data(
../astropy/astropy/coordinates/sky_coordinate_parsers.py:284: in _parse_coordinate_data
    _skycoord_kwargs, _components = _parse_coordinate_arg(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

coords = <UnitSphericalRepresentation (lon, lat) in deg
    [(nan, nan), (nan, nan), (nan, nan), ..., (nan, nan), (nan, nan),
     (nan, nan)]>
frame = <HeliographicCarrington Frame (obstime=2017-01-27T16:54:20.000, observer=<HeliographicStonyhurst Coordinate (obstime=2017-01-27T16:54:20.000): (lon, lat, radius) in (deg, deg, AU)
    (0., -5.70644239, 0.98482348)>)>
units = [None, None, None], init_kwargs = {}

    def _parse_coordinate_arg(coords, frame, units, init_kwargs):
        """
        Single unnamed arg supplied.  This must be:
        - Coordinate frame with data
        - Representation
        - SkyCoord
        - List or tuple of:
          - String which splits into two values
          - Iterable with two values
          - SkyCoord, frame, or representation objects.
    
        Returns a dict mapping coordinate attribute names to values (or lists of
        values)
        """
        from .sky_coordinate import SkyCoord
    
        is_scalar = False  # Differentiate between scalar and list input
        # valid_kwargs = {}  # Returned dict of lon, lat, and distance (optional)
        components = {}
        skycoord_kwargs = {}
    
        frame_attr_names = list(frame.representation_component_names.keys())
        repr_attr_names = list(frame.representation_component_names.values())
        repr_attr_classes = list(frame.representation_type.attr_classes.values())
        n_attr_names = len(repr_attr_names)
    
        # Turn a single string into a list of strings for convenience
        if isinstance(coords, str):
            is_scalar = True
            coords = [coords]
    
        if isinstance(coords, (SkyCoord, BaseCoordinateFrame)):
            # Note that during parsing of `frame` it is checked that any coordinate
            # args have the same frame as explicitly supplied, so don't worry here.
    
            if not coords.has_data:
                raise ValueError('Cannot initialize from a frame without coordinate data')
    
            data = coords.data.represent_as(frame.representation_type)
    
            values = []  # List of values corresponding to representation attrs
            repr_attr_name_to_drop = []
            for repr_attr_name in repr_attr_names:
                # If coords did not have an explicit distance then don't include in initializers.
                if (isinstance(coords.data, UnitSphericalRepresentation) and
                        repr_attr_name == 'distance'):
                    repr_attr_name_to_drop.append(repr_attr_name)
                    continue
    
                # Get the value from `data` in the eventual representation
                values.append(getattr(data, repr_attr_name))
    
            # drop the ones that were skipped because they were distances
            for nametodrop in repr_attr_name_to_drop:
                nameidx = repr_attr_names.index(nametodrop)
                del repr_attr_names[nameidx]
                del units[nameidx]
                del frame_attr_names[nameidx]
                del repr_attr_classes[nameidx]
    
            if coords.data.differentials and 's' in coords.data.differentials:
                orig_vel = coords.data.differentials['s']
                vel = coords.data.represent_as(frame.representation_type, frame.get_representation_cls('s')).differentials['s']
                for frname, reprname in frame.get_representation_component_names('s').items():
                    if (reprname == 'd_distance' and
                            not hasattr(orig_vel, reprname) and
                            'unit' in orig_vel.get_name()):
                        continue
                    values.append(getattr(vel, reprname))
                    units.append(None)
                    frame_attr_names.append(frname)
                    repr_attr_names.append(reprname)
                    repr_attr_classes.append(vel.attr_classes[reprname])
    
            for attr in frame_transform_graph.frame_attributes:
                value = getattr(coords, attr, None)
                use_value = (isinstance(coords, SkyCoord) or
                             attr not in coords._attr_names_with_defaults)
                if use_value and value is not None:
                    skycoord_kwargs[attr] = value
    
        elif isinstance(coords, BaseRepresentation):
            if coords.differentials and 's' in coords.differentials:
                diffs = frame.get_representation_cls('s')
                data = coords.represent_as(frame.representation_type, diffs)
                values = [getattr(data, repr_attr_name) for repr_attr_name in repr_attr_names]
                for frname, reprname in frame.get_representation_component_names('s').items():
                    values.append(getattr(data.differentials['s'], reprname))
                    units.append(None)
                    frame_attr_names.append(frname)
                    repr_attr_names.append(reprname)
                    repr_attr_classes.append(data.differentials['s'].attr_classes[reprname])
    
            else:
                data = coords.represent_as(frame.representation_type)
                values = [getattr(data, repr_attr_name) for repr_attr_name in repr_attr_names]
    
        elif (isinstance(coords, np.ndarray) and coords.dtype.kind in 'if' and
              coords.ndim == 2 and coords.shape[1] <= 3):
            # 2-d array of coordinate values.  Handle specially for efficiency.
            values = coords.transpose()  # Iterates over repr attrs
    
        elif isinstance(coords, (Sequence, np.ndarray)):
            # Handles list-like input.
    
            vals = []
            is_ra_dec_representation = ('ra' in frame.representation_component_names and
                                        'dec' in frame.representation_component_names)
            coord_types = (SkyCoord, BaseCoordinateFrame, BaseRepresentation)
            if any(isinstance(coord, coord_types) for coord in coords):
                # this parsing path is used when there are coordinate-like objects
                # in the list - instead of creating lists of values, we create
                # SkyCoords from the list elements and then combine them.
                scs = [SkyCoord(coord, **init_kwargs) for coord in coords]
    
                # Check that all frames are equivalent
                for sc in scs[1:]:
                    if not sc.is_equivalent_frame(scs[0]):
                        raise ValueError("List of inputs don't have equivalent "
                                         "frames: {} != {}".format(sc, scs[0]))
    
                # Now use the first to determine if they are all UnitSpherical
                allunitsphrepr = isinstance(scs[0].data, UnitSphericalRepresentation)
    
                # get the frame attributes from the first coord in the list, because
                # from the above we know it matches all the others.  First copy over
                # the attributes that are in the frame itself, then copy over any
                # extras in the SkyCoord
                for fattrnm in scs[0].frame.frame_attributes:
                    skycoord_kwargs[fattrnm] = getattr(scs[0].frame, fattrnm)
                for fattrnm in scs[0]._extra_frameattr_names:
                    skycoord_kwargs[fattrnm] = getattr(scs[0], fattrnm)
    
                # Now combine the values, to be used below
                values = []
                for data_attr_name, repr_attr_name in zip(frame_attr_names, repr_attr_names):
                    if allunitsphrepr and repr_attr_name == 'distance':
                        # if they are *all* UnitSpherical, don't give a distance
                        continue
                    data_vals = []
                    for sc in scs:
                        data_val = getattr(sc, data_attr_name)
                        data_vals.append(data_val.reshape(1,) if sc.isscalar else data_val)
                    concat_vals = np.concatenate(data_vals)
                    # Hack because np.concatenate doesn't fully work with Quantity
                    if isinstance(concat_vals, u.Quantity):
                        concat_vals._unit = data_val.unit
                    values.append(concat_vals)
            else:
                # none of the elements are "frame-like"
                # turn into a list of lists like [[v1_0, v2_0, v3_0], ... [v1_N, v2_N, v3_N]]
                for coord in coords:
                    if isinstance(coord, str):
                        coord1 = coord.split()
                        if len(coord1) == 6:
                            coord = (' '.join(coord1[:3]), ' '.join(coord1[3:]))
                        elif is_ra_dec_representation:
                            coord = _parse_ra_dec(coord)
                        else:
                            coord = coord1
                    vals.append(coord)  # Assumes coord is a sequence at this point
    
                # Do some basic validation of the list elements: all have a length and all
                # lengths the same
                try:
                    n_coords = sorted(set(len(x) for x in vals))
                except Exception:
                    raise ValueError('One or more elements of input sequence does not have a length')
    
                if len(n_coords) > 1:
                    raise ValueError('Input coordinate values must have same number of elements, found {}'
                                     .format(n_coords))
                n_coords = n_coords[0]
    
                # Must have no more coord inputs than representation attributes
                if n_coords > n_attr_names:
                    raise ValueError('Input coordinates have {} values but '
                                     'representation {} only accepts {}'
                                     .format(n_coords,
                                             frame.representation_type.get_name(),
                                             n_attr_names))
    
                # Now transpose vals to get [(v1_0 .. v1_N), (v2_0 .. v2_N), (v3_0 .. v3_N)]
                # (ok since we know it is exactly rectangular).  (Note: can't just use zip(*values)
                # because Longitude et al distinguishes list from tuple so [a1, a2, ..] is needed
                # while (a1, a2, ..) doesn't work.
                values = [list(x) for x in zip(*vals)]
    
                if is_scalar:
                    values = [x[0] for x in values]
        else:
            raise ValueError('Cannot parse coordinates from first argument')
    
        # Finally we have a list of values from which to create the keyword args
        # for the frame initialization.  Validate by running through the appropriate
        # class initializer and supply units (which might be None).
        try:
            for frame_attr_name, repr_attr_class, value, unit in zip(
                    frame_attr_names, repr_attr_classes, values, units):
                components[frame_attr_name] = repr_attr_class(value, unit=unit,
                                                              copy=False)
        except Exception as err:
>           raise ValueError('Cannot parse first argument data "{}" for attribute '
                             '{}'.format(value, frame_attr_name), err)
E           ValueError: ('Cannot parse first argument data "[nan nan nan ... nan nan nan] deg" for attribute lon', DeprecationWarning('Numpy has detected that you (may be) writing to an array with\noverlapping memory from np.broadcast_arrays. If this is intentional\nset the WRITEABLE flag True or make a copy immediately before writing.'))

../astropy/astropy/coordinates/sky_coordinate_parsers.py:566: ValueError
```

--- a/sunpy/map/tests/test_plotting.py
+++ b/sunpy/map/tests/test_plotting.py
@@ -264,8 +264,8 @@ def test_heliographic_quadrangle_top_right(heliographic_test_map):
     heliographic_test_map.draw_quadrangle(bottom_left, top_right=top_right, edgecolor='cyan')
 
 
-# See https://github.com/sunpy/sunpy/issues/4294 to track this warning. Ideally
-# it should not be filtered, and the cause of it fixed.
+# This warning filter can be removed once we depend on Astropy 5.0+.
+# See https://github.com/sunpy/sunpy/issues/4294
 @pytest.mark.filterwarnings(r'ignore:Numpy has detected that you \(may be\) writing to an array with\noverlapping memory')
 @figure_test
 def test_heliographic_grid_annotations(heliographic_test_map):

